backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphlAdminFieldFactory.java

```diff
@@ -80,19 +86,26 @@ private static Object getUsers(SelectedField selectedField, Database db) {
     int limit = args.containsKey(LIMIT) ? (int) args.get(LIMIT) : 100;
     int offset = args.containsKey(OFFSET) ? (int) args.get(OFFSET) : 0;
     String email = args.containsKey(EMAIL) ? (String) args.get(EMAIL) : null;
+    List<Member> members = MetadataUtils.loadUserRoles((SqlDatabase) db);
+
     if (email != null) {
-      return List.of(toGraphqlUser(db.getUser(email)));
+      return List.of(toGraphqlUser(db.getUser(email), members));
     } else {
-      return db.getUsers(limit, offset).stream()
-          .map(GraphlAdminFieldFactory::toGraphqlUser)
-          .toList();
+      return db.getUsers(limit, offset).stream().map(user -> toGraphqlUser(user, members)).toList();
     }
   }
 
-  private static Map<String, Object> toGraphqlUser(User user) {
+  private static Map<String, Object> toGraphqlUser(User user, List<Member> members) {
     Map<String, Object> result = new LinkedHashMap<>();
     result.put(EMAIL, user.getUsername());
     result.put(SETTINGS, mapSettingsToGraphql(user.getSettings()));
+
+    List<String> roles =
+        members.stream()
```

Comment:
aaaah, now I understand. You get all the memberschips and then you filter them to map them to each user.